app/models/course_template.rb

```diff
@@ -37,4 +43,35 @@ def valid_source_url?
   def clonable?
     !hidden && (expires_at.nil? || expires_at > Time.now)
   end
+
+  def cache_path
+    "#{Course.cache_root}/#{name}-#{cache_version}"
+  end
+
+  def refresh
+    firstcourse = true
+    courses.each do |c|
+      c.refresh no_directory_changes: !firstcourse
+      reload
+      firstcourse = false
+    end
```

Comment:
Yeah - I would consider it to be acceptable if we force each course to depend on course_template. but yes as long as it gets done, it doesn't need to be part of this PR.